sycl/test/extensions/annotated_usm/annotated_usm_negative.cpp

```diff
@@ -0,0 +1,87 @@
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid compile-time property in the property list.}}
+  TEST(malloc_device_annotated, N, q, properties{conduit, bar})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp:* {{static assertion failed due to requirement {{.+}}: Property is invalid for the given type.}}
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid compile-time property in the property list.}}
+  TEST(malloc_shared_annotated, N, q, properties{conduit, baz<1>})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp:* {{static assertion failed due to requirement {{.+}}: Property is invalid for the given type.}}
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid compile-time property in the property list.}}
+  TEST(malloc_host_annotated, N, q, properties{conduit, boo<double>})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp:* {{static assertion failed due to requirement {{.+}}: Property is invalid for the given type.}}
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid compile-time property in the property list.}}
+  TEST(malloc_annotated, N, q, alloc::device, properties{conduit, bar, baz<1>})
+}
+
+void testMissingUsmKind(sycl::queue &q) {
+  // missing usm kind in property list when it is required
+  properties InP{};
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_base.hpp:* {{static assertion failed due to requirement {{.+}}: USM kind is not specified. Please specify it as an argument or in the input property list.}}
+  TEST(malloc_annotated, N, q, InP)
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_base.hpp:* {{static assertion failed due to requirement {{.+}}: USM kind is not specified. Please specify it as an argument or in the input property list.}}
+  TEST((malloc_annotated<int>), N, q, InP)
+}
+
+
+void testConfilictingUsmKind(sycl::queue &q) {
+  // Conflict usm kinds between function name and property list
+  properties InP{usm_kind_host};
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Input property list contains conflicting USM kind.}}
+  // expected-error@+1 {{no matching function for call to 'malloc_shared_annotated'}}
+  auto AP = malloc_shared_annotated<int, decltype(InP)>(N, q);
```

Comment:
Why is only a single function tested here? 
Can you add tests for all the annotated malloc functions (that have the usm kind in their name)?

Reply:
test cases added